test/remote/gateways/remote_sum_up_test.rb

```diff
@@ -0,0 +1,112 @@
+require 'test_helper'
+
+class RemoteSumUpTest < Test::Unit::TestCase
+  def setup
+    @gateway = SumUpGateway.new(fixtures(:sum_up))
+
+    @amount = 100
+    @credit_card = credit_card('4000100011112224')
+    @declined_card = credit_card('55555555555555555')
+    @options = {
+      payment_type: 'card',
+      billing_address: address,
+      description: 'Store Purchase',
+      order_id: SecureRandom.uuid
+    }
+  end
+
+  def test_handle_credentials_error
+    gateway = SumUpGateway.new({ token: 'sup_sk_xx', pay_to_email: 'example@example.com' })
+    response = gateway.purchase(@amount, @visa_card, @options)
+
+    assert_equal('invalid access token', response.message)
+  end
+
+  def test_handle_pay_to_email_credential_error
+    gateway = SumUpGateway.new(fixtures(:sum_up).merge(pay_to_email: 'example@example.com'))
+    response = gateway.purchase(@amount, @visa_card, @options)
+
+    assert_equal('Validation error', response.message)
+  end
+
+  def test_successful_purchase
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success response
+    assert_equal 'PENDING', response.message
+    assert_equal @options[:order_id], response.params['checkout_reference']
+    refute_empty response.params['id']
+    refute_empty response.params['transactions']
+    refute_empty response.params['transactions'].first['id']
+    assert_equal 'PENDING', response.params['transactions'].first['status']
+  end
+
+  def test_successful_purchase_with_existing_checkout
```

Comment:
❓ Question:

Ohhhh this is super interesting my assumption was that for every authorize/purchase transaction would exist just one Checkout (given the checkout definition for SumUp), but it is allowing you to use an "already" used reference for another purchase.

Is that a reasonable use case ?

Reply:
Yes, in SumUp when a checkout is created it can have several transactions, in fact when the checkout is created a payment method is not placed. I will ask...